app/services/art_service/reports/pepfar/tx_curr_mmd.rb

```diff
@@ -0,0 +1,177 @@
+# rubocop:disable Metrics/AbcSize, Metrics/CyclomaticComplexity, Metrics/MethodLength, Metrics/PerceivedComplexity, Metrics/ClassLength, Style/Documentation
+# frozen_string_literal: true
+
+module ArtService
+  module Reports
+    module Pepfar
+      class TxCurrMmd
+        include ModelUtils
+        include Pepfar::Utils
+        include CommonSqlQueryUtils
+
+        def initialize(start_date:, end_date:, **kwargs)
+          @start_date = start_date.to_date.strftime('%Y-%m-%d 00:00:00')
+          @end_date = end_date.to_date.strftime('%Y-%m-%d 23:59:59')
+          @org = kwargs[:definition]
+          @rebuild = kwargs[:rebuild]&.casecmp?('true')
+          @occupation = kwargs[:occupation]
+          @report = init_report
+        end
+
+        def init_report
+          pepfar_age_groups.each_with_object({}) do |age_group, report|
+            report[age_group] = %w[Male Female].each_with_object({}) do |gender, age_group_report|
+              age_group_report[gender] = {
+                less_than_three_months: [],
+                three_to_five_months: [],
+                greater_than_six_months: []
+              }
+            end
+          end
+        end
+
+        private
+
+        def find_report
+          arv_concept_set = ConceptName.find_by(name: 'ARVS').concept_id
+
+          if @rebuild
+            report_type = (@org.match(/pepfar/i) ? 'pepfar' : 'moh')
+            ArtService::Reports::CohortBuilder\
+              .new(outcomes_definition: report_type)\
+              .init_temporary_tables(@start_date, @end_date, @occupation)
+          end
+
+          patients = ActiveRecord::Base.connection.select_all <<~SQL
```

Comment:
All this code can be simplified as below

```ruby
SELECT tesd.patient_id, pid.identifier arv_number, tesd.gender, tesd.birthdate,
       disaggregated_age_group(tesd.birthdate, '2024-03-31') age_group,
       TIMESTAMPDIFF(DAY, tcm.start_date, MIN(tcm.expiry_date)) prescribed_days,
       TIMESTAMPDIFF(MONTH, tcm.start_date, MIN(tcm.expiry_date)) prescribed_months
FROM temp_earliest_start_date tesd
INNER JOIN temp_patient_outcomes tpo ON tpo.patient_id = tesd.patient_id AND tpo.cum_outcome = 'On antiretrovirals'
INNER JOIN temp_current_medication tcm ON tcm.patient_id = tesd.patient_id
LEFT JOIN patient_identifier pid ON pid.patient_id = tesd.patient_id AND pid.identifier_type = 4 and pid.voided = 0
WHERE tesd.date_enrolled <= '2024-03-31'
GROUP BY tesd.patient_id;
```

After you get the result set, what you will do is just feed your response structure. You don't need to fetching drugs, weight, etc again.

You will need to replace the date in the query with the end date